front/lib/api/assistant/actions/dust_app_run.ts

```diff
@@ -164,17 +164,54 @@ export class DustAppRunConfigurationServerRunner extends BaseActionConfiguration
       }
     }
 
+    if (name !== app.name) {
+      return new Err(
+        new Error("Unreachable: Dust app name and action name differ.")
+      );
+    }
+
     return dustAppRunActionSpecification({
       schema,
-      name: name ?? app.name,
-      description: description ?? app.description ?? "",
+      name,
+      description: description ?? app.description ?? `Run app ${app.name}`,
     });
   }
 
   async deprecatedBuildSpecificationForSingleActionAgent(
     auth: Authenticator
   ): Promise<Result<AgentActionSpecification, Error>> {
-    return this.buildSpecification(auth, {});
+    // We need to pull the app comply to the required name signature while not having a name in
```

Comment:
Temporary duplicate app retrieval that will disappear in my next PR where I remove entirely all `deprecatedBuildSpecificationForSingleActionAgent`